great_expectations/data_context/types/base.py

```diff
@@ -558,6 +558,17 @@ def validate_schema(self, data, **kwargs):
 continue.
                     """
             )
+        if "gcs_options" in data and data["class_name"] in [
+            "InferredAssetGCSDataConnector",
+            "ConfiguredAssetGCSDataConnector",
+        ]:
+            azure_options = data["gcs_options"]
+            if "filename" in azure_options and "info" in azure_options:
+                raise ge_exceptions.InvalidConfigError(
+                    f"""Your current configuration can only use a single method of authentication for the GCS type of data connector.
+                    You must only select one between `filename` (from_service_account_file) and `info` (from_service_account_info). Please update your configuration to continue.
+                    """
+                )
```

Comment:
We are supporting the two primary methods of instantiating a `Credential` Google SDK object:
1. `from_service_account_info` (takes `info` as an arg)
2. `from_service_account_file` (takes `filename` as an arg)
https://google-auth.readthedocs.io/en/master/reference/google.oauth2.service_account.html

Reply:
https://googleapis.dev/python/google-api-core/latest/auth.html details the various auth methods supported by the SDK

Reply:
You can review the actual implementation that uses these changes: https://github.com/great-expectations/great_expectations/pull/3247/files